src/overlays/actors/ovl_En_Part/z_en_part.c

```diff
@@ -21,23 +26,307 @@ const ActorInit En_Part_InitVars = {
+
+    Player* player = PLAYER;
+    Vec3f velocity;
+    Vec3f accel;
+    u8 invincibilityTimer;
+
+    this->effectDelay--;
+    if (this->effectDelay == 0) {
+        Actor_Kill(&this->actor);
+        return;
+    }
+
+    velocity = D_80ACF1D4;
+    accel = D_80ACF1E0;
+
+    if (sqrt(this->actor.xyzDistFromLinkSq) <= 40.0f) {
+        invincibilityTimer = player->invincibilityTimer;
+        if (player->invincibilityTimer <= 0) {
+            if (player->invincibilityTimer <= -40) {
+                player->invincibilityTimer = 0;
+            } else {
+                player->invincibilityTimer = 0;
+                globalCtx->damagePlayer(globalCtx, -8);
+            }
+        }
+        func_8002F71C(globalCtx, this->actor.parent, (650.0f - this->actor.parent->xzDistFromLink) * 0.04f + 4.0f,
+                      this->actor.parent->posRot.rot.y, 8.0f);
+        player->invincibilityTimer = invincibilityTimer;
+        this->effectDelay = 1;
+    }
+
+    func_80033480(globalCtx, &this->actor.posRot.pos, 0.0f, 1, 300, 150, 1);
+    velocity.x = Math_Rand_CenteredFloat(16.0f);
+    func_80029724(globalCtx, &this->actor.posRot.pos, &velocity, &accel, 20,
+                  (s32)((Math_Rand_ZeroOne() * 5.0f + 12.0f) * 2), -1, 10, 0);
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_MONBLIN_GNDWAVE - SFX_FLAG);
+}
+
+void func_80ACE7E8(EnPart* this, GlobalContext* globalCtx) {
+    static Vec3f D_80ACF1EC = { 0.0f, 0.0f, 0.0f };
+
+    Vec3f zero = D_80ACF1EC;
+    f32 diffsSum;
+
+    if ((this->actor.parent == NULL) || (this->actor.parent->update == NULL)) {
+        func_8002A6B8(globalCtx, &this->actor.posRot.pos, &zero, &zero, (s16)(this->actor.scale.y * 100.0f) * 40, 7,
+                      0xFF, 0xFF, 0xFF, 0xFF, 0, 0xFF, 0, 1, 9, 1);
+        Actor_Kill(&this->actor);
+        return;
+    }
+
+    if (this->effectDelay == 0) {
+        diffsSum = Math_SmoothScaleMaxMinF(&this->actor.posRot.pos.x, this->actor.initPosRot.pos.x, 1.0f, 5.0f, 0.0f);
+        diffsSum += Math_SmoothScaleMaxMinF(&this->actor.posRot.pos.y, this->actor.initPosRot.pos.y, 1.0f, 5.0f, 0.0f);
+        diffsSum += Math_SmoothScaleMaxMinF(&this->actor.posRot.pos.z, this->actor.initPosRot.pos.z, 1.0f, 5.0f, 0.0f);
+        diffsSum += Math_SmoothScaleMaxMinF(&this->rotZ, 0.0f, 1.0f, 0.25f, 0.0f);
+        if (diffsSum == 0.0f) {
+            this->actor.parent->initPosRot.rot.x--;
+            this->effectDelay--;
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_STAL_DAMAGE);
+        }
+    } else if (this->effectDelay > 0) {
+        this->effectDelay--;
+    }
+
+    if (this->actor.parent->colChkInfo.health != 0) {
+        Actor_Kill(&this->actor);
+    }
+}
+
+void EnPart_Update(Actor* thisx, GlobalContext* globalCtx) {
+    static EnPartActionFunc sActionFuncs[] = { func_80ACDDE8, func_80ACE13C, func_80ACE5B8, func_80ACE5C8,
+                                               func_80ACE7E8 };
+
+    EnPart* this = THIS;
```

Comment:
I would personally leave this array of functions outside the file (right above it), but if you want to keep it inside the function I suggest formatting it like this (note the trailing comma):
```suggestion
    static EnPartActionFunc sActionFuncs[] = {
        func_80ACDDE8, func_80ACE13C, func_80ACE5B8, func_80ACE5C8, func_80ACE7E8,
    };
    EnPart* this = THIS;
```